File: Views/Subscription/SubscriptionView.swift
import SwiftUI
import StoreKit

struct SubscriptionView: View {
    @EnvironmentObject var appState: AppState
    @StateObject private var receiptService = AppleReceiptService.shared
    @Environment(\.dismiss) private var dismiss
    @State private var showingAppleSignIn = false
    @State private var showingSuccessMessage = false
    @State private var purchaseCompleted = false
    
    var body: some View {
        ZStack {
            // Fond gris clair identique aux autres pages d'onboarding
            Color(red: 0.97, green: 0.97, blue: 0.98)
                .ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Header avec croix collée en haut
                HStack {
                    Button(action: {
                        print("🔥 SubscriptionView: Fermeture via croix")
                        appState.freemiumManager?.dismissSubscription()
                        dismiss()
                    }) {
                        Image(systemName: "xmark")
                            .font(.system(size: 18, weight: .medium))
                            .foregroundColor(.black)
                    }
                    .padding(.leading, 20)
                    
                    Spacer()
                }
                .padding(.top, 10)
                
                // Titre principal collé sous la croix
                VStack(spacing: 8) {
                    Text("choose_plan".localized)
                        .font(.system(size: 32, weight: .bold))
                        .foregroundColor(.black)
                        .multilineTextAlignment(.center)
                    
                    // Sous-titre
                    Text("partner_no_payment".localized)
                        .font(.system(size: 16))
                        .foregroundColor(.black.opacity(0.7))
                        .multilineTextAlignment(.center)
                }
                .padding(.horizontal, 20)
                .padding(.top, 15)
                
                Spacer()
                    .frame(height: 20)
                
                // Contenu principal - Nouvelles fonctionnalités
                VStack(spacing: 25) {
                    
                    NewFeatureRow(
                        title: "love_stronger_feature".localized,
                        subtitle: "love_stronger_description".localized
                    )

                    NewFeatureRow(
                        title: "memory_box_feature".localized,
                        subtitle: "memory_box_description".localized
                    )

                    NewFeatureRow(
                        title: "love_map_feature".localized,
                        subtitle: "love_map_description".localized
                    )
                }
                .padding(.horizontal, 25)
                
                Spacer()
                
                // Garder l'espace entre les boutons de choix et le bouton commencer
                VStack(spacing: 0) {
                    // Section de sélection des plans - ORDRE INVERSÉ
                    VStack(spacing: 8) {
                        // Plan Hebdomadaire (maintenant en premier)
                        PlanSelectionCard(
                            planType: .weekly,
                            isSelected: receiptService.selectedPlan == .weekly,
                            onTap: {
                                receiptService.selectedPlan = .weekly
                            }
                        )
                        
                        // Plan Mensuel (maintenant en second) sans badge externe
                        PlanSelectionCard(
                            planType: .monthly,
                            isSelected: receiptService.selectedPlan == .monthly,
                            onTap: {
                                receiptService.selectedPlan = .monthly
                            }
                        )
                    }
                    .padding(.horizontal, 20)
                    .padding(.bottom, 3)
                    
                    Spacer()
                        .frame(height: 18)
                    
                    // Texte informatif - AU-DESSUS DU BOUTON
                    HStack(spacing: 5) {
                        Image(systemName: "checkmark")
                            .font(.system(size: 14, weight: .bold))
                            .foregroundColor(.black)
                        
                        Text(receiptService.selectedPlan == .monthly ? 
                                             "no_payment_required_now".localized :
                "no_commitment_cancel_anytime".localized)
                            .font(.system(size: 14))
                            .foregroundColor(.black)
                    }
                    .padding(.bottom, 12)
                    
                    // Bouton d'achat
                    Button(action: {
                        purchaseSubscription()
                    }) {
                        HStack {
                            if receiptService.isLoading {
                                HStack(spacing: 8) {
                                    ProgressView()
                                        .progressViewStyle(CircularProgressViewStyle(tint: .white))
                                        .scaleEffect(0.8)
                                    Text("loading_caps".localized)
                                }
                            } else {
                                Text("start_trial".localized)
                            }
                        }
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .frame(height: 56)
                        .background(Color(hex: "#FD267A"))
                        .cornerRadius(28)
                    }
                    .disabled(receiptService.isLoading)
                    .padding(.horizontal, 30)
                    
                    Spacer()
                        .frame(height: 12)
                    
                    // Liens légaux et restaurer tout en bas de l'écran (design moderne)
                    HStack(spacing: 15) {
                        Button("terms".localized) {
                            if let url = URL(string: "https://www.apple.com/legal/internet-services/itunes/dev/stdeula/") {
                                UIApplication.shared.open(url)
                            }
                        }
                        .font(.system(size: 12))
                        .foregroundColor(.black.opacity(0.5))
                        
                        Button("privacy_policy".localized) {
                            if let url = URL(string: "https://love2lovesite.onrender.com") {
                                UIApplication.shared.open(url)
                            }
                        }
                        .font(.system(size: 12))
                        .foregroundColor(.black.opacity(0.5))
                        
                        Button("restore".localized) {
                            print("🔥 SubscriptionView: Tentative de restauration des achats")
                            receiptService.restorePurchases()
                        }
                        .font(.system(size: 12))
                        .foregroundColor(.black.opacity(0.5))
                    }
                    .padding(.bottom, 5)
                }
            }
            
            // Affichage des erreurs
            if let errorMessage = receiptService.errorMessage {
                VStack {
                    Spacer()
                    Text(errorMessage)
                        .font(.system(size: 14))
                        .foregroundColor(.white)
                        .padding()
                        .background(Color.red.opacity(0.8))
                        .cornerRadius(10)
                        .padding(.horizontal, 30)
                        .padding(.bottom, 100)
                }
            }
        }
        .navigationBarHidden(true)
        .onAppear {
            print("🔥 SubscriptionView: Vue apparue")
            // Analytics - tracker l'affichage de la vue subscription
            appState.freemiumManager?.trackUpgradePromptShown()
        }
        .onReceive(receiptService.$isSubscribed) { isSubscribed in
            print("🔥 SubscriptionView: Changement d'état d'abonnement: \(isSubscribed)")
            if isSubscribed {
                // L'utilisateur s'est abonné, mettre à jour Firebase et fermer
                print("🔥 SubscriptionView: Abonnement validé - mise à jour Firebase")
                
                // Mettre à jour l'utilisateur dans Firebase
                if var currentUser = appState.currentUser {
                    currentUser.isSubscribed = true
                    appState.updateUser(currentUser)
                }
                
                // Analytics
                appState.freemiumManager?.trackConversion()
                
                // Fermer la vue
                appState.freemiumManager?.dismissSubscription()
                dismiss()
            }
        }
        .onReceive(receiptService.$errorMessage) { errorMessage in
            if let error = errorMessage {
                print("🔥 SubscriptionView: Erreur reçue: \(error)")
            }
        }
    }
    
    private func purchaseSubscription() {
        print("🔥 SubscriptionView: Tentative d'achat")
        // Analytics - tracker tentative d'achat
        appState.freemiumManager?.trackUpgradePromptShown()
        receiptService.purchaseSubscription()
    }
}

// Composant pour les cartes de sélection de plan - VERSION MODIFIÉE
struct PlanSelectionCard: View {
    let planType: SubscriptionPlanType
    let isSelected: Bool
    let onTap: () -> Void
    
    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 15) {
                // Contenu du plan
                VStack(alignment: .leading, spacing: 5) {
                    HStack {
                        Text(planType.displayName)
                            .font(.system(size: 17, weight: .bold))
                            .foregroundColor(.black)
                        
                        Spacer()
                    }
                    
                    HStack(spacing: 0) {
                        Text("\(planType.price) / \(planType.period)")
                            .font(.system(size: 13, weight: .bold))
                            .foregroundColor(.black)
                        Text("for_2_users".localized)
                            .font(.system(size: 13, weight: .regular))
                            .foregroundColor(.black)
                    }
                    
                                            Text("\(planType.pricePerUser) " + "per_user_per".localized + " \(planType.period)")
                        .font(.system(size: 12))
                        .foregroundColor(.black.opacity(0.7))
                }
                
                Spacer()
                
                // Coche de sélection à droite
                ZStack {
                    Circle()
                        .stroke(Color.black, lineWidth: 2)
                        .frame(width: 24, height: 24)
                    
                    if isSelected {
                        Circle()
                            .fill(Color.black)
                            .frame(width: 24, height: 24)
                        
                        Image(systemName: "checkmark")
                            .font(.system(size: 14, weight: .bold))
                            .foregroundColor(.white)
                    }
                }
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 8)
            .background(
                RoundedRectangle(cornerRadius: 12)
                    .fill(Color.white)
                    .overlay(
                        RoundedRectangle(cornerRadius: 12)
                            .stroke(isSelected ? Color.black : Color.black.opacity(0.3), lineWidth: isSelected ? 2 : 1)
                    )
            )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

// Composant PremiumFeatureRow - pour compatibilité si utilisé ailleurs
struct PremiumFeatureRow: View {
    let title: String
    let subtitle: String
    
    var body: some View {
        NewFeatureRow(title: title, subtitle: subtitle)
    }
}

// Composant NewFeatureRow - design moderne
struct NewFeatureRow: View {
    let title: String
    let subtitle: String
    
    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(title)
                .font(.system(size: 19, weight: .bold))
                .foregroundColor(.black)
                .multilineTextAlignment(.leading)
            
            Text(subtitle)
                .font(.system(size: 15))
                .foregroundColor(.black.opacity(0.7))
                .multilineTextAlignment(.leading)
                .lineLimit(nil)
                .fixedSize(horizontal: false, vertical: true)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
} 

File: Views/Subscription/SubscriptionView.kt
// package com.yourapp.ui.subscription

/*
 * Kotlin/Compose port of the SwiftUI SubscriptionView.
 * ✅ Uses Android strings.xml (via stringResource) instead of .xcstrings.
 * ✅ Mirrors layout, plan selection order, and analytics hooks.
 * ✅ Hooks into your existing services (AppState, Receipt/Pricing services).
 *
 * Required strings.xml keys (create them if missing):
 *  - choose_plan
 *  - partner_no_payment
 *  - love_stronger_feature
 *  - love_stronger_description
 *  - memory_box_feature
 *  - memory_box_description
 *  - love_map_feature
 *  - love_map_description
 *  - no_payment_required_now
 *  - no_commitment_cancel_anytime
 *  - loading_caps
 *  - start_trial
 *  - terms
 *  - privacy_policy
 *  - restore
 *  - for_2_users
 *  - per_user_per
 */

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.LocalLifecycleOwner
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleEventObserver
import com.google.firebase.analytics.ktx.analytics
import com.google.firebase.ktx.Firebase
import kotlinx.coroutines.flow.StateFlow

// ---------- Placeholders / contracts ----------
// These mimic your iOS services. Wire them to your Android implementations.
// If you already have equivalents, keep only the Composables and remove these.

class AppState(
    var currentUser: User? = null,
    val freemiumManager: FreemiumManager? = null
) {
    fun updateUser(user: User) { /* TODO: implement */ }
}

data class User(val isSubscribed: Boolean)

interface FreemiumManager {
    fun dismissSubscription()
    fun trackUpgradePromptShown()
    fun trackConversion()
}

enum class SubscriptionPlanType { WEEKLY, MONTHLY }

data class PlanInfo(
    val displayName: String,
    val price: String,
    val period: String,
    val pricePerUser: String
)

interface PlayBillingPricingService {
    val hasDynamicPrices: StateFlow<Boolean>
    fun refreshPrices()
    fun infoFor(plan: SubscriptionPlanType): PlanInfo
}

interface GoogleReceiptService {
    val isLoading: StateFlow<Boolean>
    val isSubscribed: StateFlow<Boolean>
    val errorMessage: StateFlow<String?>
    val selectedPlan: StateFlow<SubscriptionPlanType>

    fun selectPlan(plan: SubscriptionPlanType)
    fun purchaseSubscription()
    fun restorePurchases()
}

// ---------- Color helpers ----------
private fun hex(hex: String): Color = Color(android.graphics.Color.parseColor(hex))

// ---------- Composable ----------
@Composable
fun SubscriptionScreen(
    appState: AppState,
    receiptService: GoogleReceiptService,
    pricingService: PlayBillingPricingService,
    onDismiss: () -> Unit
) {
    val context = LocalContext.current

    // Collect service state
    val isLoading by receiptService.isLoading.collectAsState(initial = false)
    val isSubscribed by receiptService.isSubscribed.collectAsState(initial = false)
    val selectedPlan by receiptService.selectedPlan.collectAsState(initial = SubscriptionPlanType.WEEKLY)
    val errorMessage by receiptService.errorMessage.collectAsState(initial = null)
    val hasDynamicPrices by pricingService.hasDynamicPrices.collectAsState(initial = false)

    // On appear: refresh prices if needed + analytics view shown
    LaunchedEffect(hasDynamicPrices) {
        if (!hasDynamicPrices) pricingService.refreshPrices()
    }
    // Mirror SwiftUI .onAppear analytics
    LaunchedEffect(Unit) {
        appState.freemiumManager?.trackUpgradePromptShown()
    }

    // Observe subscription success (like Swift .onReceive)
    LaunchedEffect(isSubscribed) {
        if (isSubscribed) {
            appState.currentUser?.let { current ->
                appState.updateUser(current.copy(isSubscribed = true))
            }
            // Firebase Analytics events
            Firebase.analytics.logEvent("abonnement_reussi", null)
            appState.freemiumManager?.trackConversion()
            appState.freemiumManager?.dismissSubscription()
            onDismiss()
        }
    }

    // UI
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF7F7F9)) // same light grey backdrop
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
        ) {
            // Header with close button (top-aligned)
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = 10.dp, start = 20.dp, end = 20.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                IconButton(onClick = {
                    appState.freemiumManager?.dismissSubscription()
                    onDismiss()
                }) {
                    Icon(
                        imageVector = Icons.Default.Close,
                        contentDescription = "Close",
                        tint = Color.Black
                    )
                }
                Spacer(modifier = Modifier.weight(1f))
            }

            // Title + subtitle
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp)
                    .padding(top = 15.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text(
                    text = stringResource(id = R.string.choose_plan),
                    fontSize = 32.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black,
                    textAlign = TextAlign.Center
                )
                Spacer(Modifier.height(8.dp))
                Text(
                    text = stringResource(id = R.string.partner_no_payment),
                    fontSize = 16.sp,
                    color = Color.Black.copy(alpha = 0.7f),
                    textAlign = TextAlign.Center
                )
            }

            Spacer(Modifier.height(20.dp))

            // Feature bullets
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 25.dp),
                verticalArrangement = Arrangement.spacedBy(25.dp)
            ) {
                NewFeatureRow(
                    title = stringResource(R.string.love_stronger_feature),
                    subtitle = stringResource(R.string.love_stronger_description)
                )
                NewFeatureRow(
                    title = stringResource(R.string.memory_box_feature),
                    subtitle = stringResource(R.string.memory_box_description)
                )
                NewFeatureRow(
                    title = stringResource(R.string.love_map_feature),
                    subtitle = stringResource(R.string.love_map_description)
                )
            }

            Spacer(modifier = Modifier.weight(1f))

            // Plans + CTA + legal
            Column(
                modifier = Modifier.fillMaxWidth(),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Selection cards (order: WEEKLY first, then MONTHLY)
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 20.dp),
                    verticalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    PlanSelectionCard(
                        planType = SubscriptionPlanType.WEEKLY,
                        isSelected = selectedPlan == SubscriptionPlanType.WEEKLY,
                        onTap = { receiptService.selectPlan(SubscriptionPlanType.WEEKLY) },
                        pricingService = pricingService
                    )
                    PlanSelectionCard(
                        planType = SubscriptionPlanType.MONTHLY,
                        isSelected = selectedPlan == SubscriptionPlanType.MONTHLY,
                        onTap = { receiptService.selectPlan(SubscriptionPlanType.MONTHLY) },
                        pricingService = pricingService
                    )
                }

                Spacer(Modifier.height(18.dp))

                // Info line above CTA
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    modifier = Modifier.padding(bottom = 12.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.Check,
                        contentDescription = null,
                        tint = Color.Black,
                        modifier = Modifier.size(16.dp)
                    )
                    Spacer(Modifier.width(5.dp))
                    val infoText = if (selectedPlan == SubscriptionPlanType.MONTHLY) {
                        stringResource(R.string.no_payment_required_now)
                    } else {
                        stringResource(R.string.no_commitment_cancel_anytime)
                    }
                    Text(
                        text = infoText,
                        fontSize = 14.sp,
                        color = Color.Black
                    )
                }

                // Purchase button
                Button(
                    onClick = {
                        val planTypeStr = if (selectedPlan == SubscriptionPlanType.WEEKLY) "weekly" else "monthly"
                        Firebase.analytics.logEvent("abonnement_demarre", Bundle().apply {
                            putString("type", planTypeStr)
                            putString("source", "subscription_screen")
                        })
                        appState.freemiumManager?.trackUpgradePromptShown()
                        receiptService.purchaseSubscription()
                    },
                    enabled = !isLoading,
                    modifier = Modifier
                        .padding(horizontal = 30.dp)
                        .fillMaxWidth()
                        .height(56.dp),
                    shape = RoundedCornerShape(28.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = hex("#FD267A"))
                ) {
                    if (isLoading) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            CircularProgressIndicator(
                                modifier = Modifier
                                    .size(18.dp),
                                color = Color.White,
                                strokeWidth = 2.dp
                            )
                            Spacer(Modifier.width(8.dp))
                            Text(
                                text = stringResource(R.string.loading_caps),
                                fontSize = 18.sp,
                                fontWeight = FontWeight.Bold,
                                color = Color.White
                            )
                        }
                    } else {
                        Text(
                            text = stringResource(R.string.start_trial),
                            fontSize = 18.sp,
                            fontWeight = FontWeight.Bold,
                            color = Color.White
                        )
                    }
                }

                Spacer(Modifier.height(12.dp))

                // Legal + Restore
                Row(
                    horizontalArrangement = Arrangement.spacedBy(15.dp),
                    verticalAlignment = Alignment.CenterVertically,
                    modifier = Modifier.padding(bottom = 5.dp)
                ) {
                    Text(
                        text = stringResource(R.string.terms),
                        fontSize = 12.sp,
                        color = Color.Black.copy(alpha = 0.5f),
                        modifier = Modifier.clickable {
                            // Using Apple EULA like iOS snippet; replace with your Terms URL if different on Android
                            openUrl(context, "https://www.apple.com/legal/internet-services/itunes/dev/stdeula/")
                        }
                    )
                    Text(
                        text = stringResource(R.string.privacy_policy),
                        fontSize = 12.sp,
                        color = Color.Black.copy(alpha = 0.5f),
                        modifier = Modifier.clickable {
                            val isFr = context.resources.configuration.locales[0]?.language?.startsWith("fr") == true
                            val privacyUrl = if (isFr) {
                                "https://love2lovesite.onrender.com"
                            } else {
                                "https://love2lovesite.onrender.com/privacy-policy.html"
                            }
                            openUrl(context, privacyUrl)
                        }
                    )
                    Text(
                        text = stringResource(R.string.restore),
                        fontSize = 12.sp,
                        color = Color.Black.copy(alpha = 0.5f),
                        modifier = Modifier.clickable {
                            receiptService.restorePurchases()
                        }
                    )
                }
            }
        }

        // Error toast (bottom)
        if (!errorMessage.isNullOrEmpty()) {
            Box(
                modifier = Modifier
                    .fillMaxSize(),
                contentAlignment = Alignment.BottomCenter
            ) {
                Text(
                    text = errorMessage ?: "",
                    color = Color.White,
                    fontSize = 14.sp,
                    modifier = Modifier
                        .padding(bottom = 100.dp)
                        .padding(horizontal = 30.dp)
                        .background(Color.Red.copy(alpha = 0.8f), RoundedCornerShape(10.dp))
                        .padding(horizontal = 16.dp, vertical = 12.dp)
                )
            }
        }
    }
}

private fun openUrl(context: android.content.Context, url: String) {
    runCatching {
        context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(url)))
    }
}

// ---------- Rows / Cards ----------
@Composable
fun NewFeatureRow(title: String, subtitle: String) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.Start
    ) {
        Text(
            text = title,
            fontSize = 19.sp,
            fontWeight = FontWeight.Bold,
            color = Color.Black,
            textAlign = TextAlign.Start
        )
        Spacer(Modifier.height(8.dp))
        Text(
            text = subtitle,
            fontSize = 15.sp,
            color = Color.Black.copy(alpha = 0.7f),
            textAlign = TextAlign.Start
        )
    }
}

@Composable
fun PremiumFeatureRow(title: String, subtitle: String) {
    NewFeatureRow(title = title, subtitle = subtitle)
}

@Composable
fun PlanSelectionCard(
    planType: SubscriptionPlanType,
    isSelected: Boolean,
    onTap: () -> Unit,
    pricingService: PlayBillingPricingService
) {
    val info = pricingService.infoFor(planType)

    Box(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.White, RoundedCornerShape(12.dp))
            .border(
                width = if (isSelected) 2.dp else 1.dp,
                color = if (isSelected) Color.Black else Color.Black.copy(alpha = 0.3f),
                shape = RoundedCornerShape(12.dp)
            )
            .clickable(onClick = onTap)
            .padding(horizontal = 20.dp, vertical = 8.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Column(modifier = Modifier.weight(1f)) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text(
                        text = info.displayName,
                        fontSize = 17.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.Black
                    )
                    Spacer(Modifier.weight(1f))
                }
                Spacer(Modifier.height(2.dp))
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text(
                        text = "${info.price} / ${info.period}",
                        fontSize = 13.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.Black
                    )
                    Spacer(Modifier.width(4.dp))
                    Text(
                        text = stringResource(R.string.for_2_users),
                        fontSize = 13.sp,
                        color = Color.Black
                    )
                }
                Spacer(Modifier.height(2.dp))
                Text(
                    text = "${info.pricePerUser} ${stringResource(R.string.per_user_per)} ${info.period}",
                    fontSize = 12.sp,
                    color = Color.Black.copy(alpha = 0.7f)
                )
            }

            // Selection circle
            Box(
                modifier = Modifier
                    .size(24.dp),
                contentAlignment = Alignment.Center
            ) {
                Box(
                    modifier = Modifier
                        .matchParentSize()
                        .border(2.dp, Color.Black, CircleShape)
                )
                if (isSelected) {
                    Box(
                        modifier = Modifier
                            .matchParentSize()
                            .background(Color.Black, CircleShape)
                    )
                    Icon(
                        imageVector = Icons.Default.Check,
                        contentDescription = null,
                        tint = Color.White,
                        modifier = Modifier.size(14.dp)
                    )
                }
            }
        }
    }
}
